release/release.py

```diff
@@ -0,0 +1,331 @@
+    control_file_template = os.path.join(metadata.cargoroot, "release/mount-s3.debian-control")
+    with open(control_file_template) as f:
+        control_file = f.read()
+    control_file = control_file.replace("__VERSION__", metadata.version_string)
+    # Debian's architecture strings don't match `uname`
+    if metadata.arch == "x86_64":
+        deb_arch = "amd64"
+    elif metadata.arch == "aarch64":
+        deb_arch = "arm64"
+    else:
+        raise Exception(f"unknown architecture {metadata.args} for DEB package")
+    control_file = control_file.replace("__ARCH__", deb_arch)
+    control_file_path = os.path.join(deb_DEBIAN_dir, "control")
+    with open(control_file_path, "w") as f:
+        f.write(control_file)
+
+    # For DEB we need to create the bin symlinks ourselves
+    deb_bin_dir = os.path.join(deb_package_dir, "usr/bin")
+    os.makedirs(deb_bin_dir)
+    os.symlink(f"/{OPT_PATH}/bin/mount-s3", os.path.join(deb_bin_dir, "mount-s3"))
+
+    # Build the DEB
+    deb_path = os.path.join(deb_buildroot, "mount-s3.deb")
+    run(["fakeroot", "dpkg-deb", "--build", "-Zxz", deb_package_dir, deb_path])
+
+    final_deb_path = os.path.join(metadata.output_dir, metadata.artifact_name("deb"))
+    shutil.copy2(os.path.join(deb_buildroot, "mount-s3.deb"), final_deb_path)
+
+    log(f"Built DEB: {final_deb_path}")
+
+    return final_deb_path
+
+
+def build_package_archive(metadata: BuildMetadata, package_dir: str) -> str:
+    """Build a .tar.gz archive from the contents of the package directory. Return the path to the
+    final .tar.gz archive."""
+
+    archive_path = os.path.join(metadata.output_dir, metadata.artifact_name("tar.gz"))
+    run(["tar", "czvf", archive_path, "-C", package_dir, "."])
+    return archive_path
+
+
+def build(args: argparse.Namespace) -> str:
+    """Top-level build driver."""
+
+    check_dependencies(args)
+    metadata = get_build_metadata(args)
+
+    binary_path = build_mountpoint_binary(metadata, args)
+    attribution_path = build_attribution(metadata)
+
+    package_dir = build_package_dir(metadata, binary_path, attribution_path)
+
+    artifacts = []
+    if not args.no_rpm:
+        artifacts.append(build_rpm(metadata, package_dir))
+    if not args.no_deb:
+        artifacts.append(build_deb(metadata, package_dir))
+    artifacts.append(build_package_archive(metadata, package_dir))
+
+    for path in artifacts:
+        os.chmod(path, 0o755)
+
+    return artifacts
+
+
+if __name__ == "__main__":
+    p = argparse.ArgumentParser()
+    p.add_argument("--root-dir", help="override the path to the Cargo workspace")
+    p.add_argument("--expected-version", help="expected version number for the Mountpoint binary")
+    p.add_argument("--no-rpm", action="store_true", help="do not build an RPM")
+    p.add_argument("--no-deb", action="store_true", help="do not build a DEB")
+    p.add_argument("--unofficial", action="store_true", help="tag the release as unofficial (for CI use)")
```

Comment:
I would prefer this to be `unofficial` by default, people could run this script on their own and miss adding the flag, result in unintentional "official" release.

Reply:
Yeah, done.